mteb/abstasks/CrosslingualTask.py

```diff
@@ -20,6 +20,31 @@ def load_data(self, **kwargs):
         """Load dataset from HuggingFace hub"""
         if self.data_loaded:
             return
+
+        fast_loading = self.fast_loading if hasattr(self, "fast_loading") else False
+        if fast_loading:
+            self.fast_load()
+        else:
+            self.slow_load()
+
+    def fast_load(self, **kwargs):
+        """Load all subsets at once, then group by language with Polars"""
+        self.dataset = {}
+        merged_dataset = datasets.load_dataset(
+            **self.metadata_dict["dataset"]
+        )  # load "default" subset
```

Comment:
This loads all the subsets and then filter the needed languages. We could load granularly only the required languages as shown here: https://github.com/huggingface/datasets/issues/6800#issuecomment-2057225862

That would require to use a standard data path for each dataset, along with handling train/test/eval splits, I feel like the overhead of downloading more data is ok since datasets shouldn't be huge anyway

Reply:
The thing is that with load_datasets you can't really select the subsets to load, what you do is to select all the files that you want to load. So if a subset "en-fr" is split into multiple files 0001.jsonl 0002.jsonl etc. we would have to account for that in the loading 

But again maybe we can just make it a constraint to format dataset with one file per split/subset